kin-devplatform-sdk/src/main/java/kin/devplatform/settings/presenter/SettingsPresenter.java

```diff
@@ -177,16 +184,48 @@ private void switchAccount(int accountIndex) {
 		accountManager.switchAccount(accountIndex, new KinCallback<Boolean>() {
 			@Override
 			public void onResponse(Boolean response) {
+				view.stopWaiting();
 				eventLogger.send(RestoreWalletCompleted.create());
 			}
 
 			@Override
 			public void onFailure(KinEcosystemException exception) {
+				view.stopWaiting();
 				eventLogger.send(GeneralEcosystemSdkError
 					.create(ErrorUtil.getPrintableStackTrace(exception), String.valueOf(exception.getCode()),
 						"SettingsPresenter.switchAccount onFailure"));
 				showCouldNotImportAccountError();
 			}
+		}, new IMigrationManagerCallbacks() {
+
+			@Override
+			public void onMigrationStart() {
+				if (view != null) {
+					// TODO: 07/02/2019 confirm with Ayelet
+					view.showMigrationStartedDialog();
+				}
+			}
+
+			@Override
+			public void onReady(IKinClient kinClient) {
+				if (view != null) {
+					// TODO: 07/02/2019 Maybe if it is not started then show different message because there were no actual migration.
```

Comment:
don't show migration related UI in case no migration happened (like the external callback to developer)

Reply:
I agree but will talk about it also like the previous comment.